packages/core/src/tracing/sentrySpan.ts

```diff
@@ -118,18 +151,21 @@ export class SentrySpan implements Span {
   }
 
   /** @inheritdoc */
-  public setAttribute(key: string, value: SpanAttributeValue | undefined): void {
+  public setAttribute(key: string, value: SpanAttributeValue | undefined): this {
```

Comment:
Technically that's a breaking change 😬 

Sure, 99% of our users will simply call `span.setAttribute` but maybe for some reason someone asserts on `undefined` being returned. Is there a concrete reason (e.g. someone running into this mismatch) we should consider breaking here? If not, I'd vote we don't change the return types of `setAttribute(s)` in v8 but happy to discuss this.

Reply:
This is not breaking unless somebody implements the SentrySpan interface, which is not really something we recommend anyone doing.

Reply:
Hmm maybe I'm missing something but this (including variations of working with the return type) seems like valid code to me:
 
```js
Sentry.startSpan({}, span => {
  assert(span.setAttribute(...) === undefined) 
})
```

I'm not saying this is a likely code path. We can make the call to technically break here. If we do, I'd prefer to extract the breaking aspect into a separate PR to revert more easily in case someone does write in.

Reply:
Widening return types is generally not considered semver major, but sever minor I believe.

Reply:
I'd say this is fine, we generally accept widening return types as non-breaking I believe, the potential for breaking here is _extremely_ low 🤔

Reply:
Ok, I'll keep the change then?

Reply:
Yeah, let's keep it.